amr-wind/immersed_boundary/IBSrcOp.H

```diff
@@ -74,42 +69,40 @@ void IBSrcOp<GeomTrait>::operator()(
     const std::string fname = GeomTrait::identifier();
     BL_PROFILE("amr-wind::IBSrcOp<" + fname + ">");
 
+    const amrex::Real dt = m_data.sim().deltaT();
+
     const auto& bx = mfi.tilebox();
-    const auto& sarr = m_ib_src(lev).array(mfi);
     const auto& problo = geom.ProbLoArray();
     const auto& dx = geom.CellSizeArray();
+    const amrex::Real dV = dx[0]*dx[1]*dx[2];
 
-    const int npts = m_pos.size();
+    // extract relevant fields
     const auto* pos = m_pos.data();
-    const auto* force = m_force.data();
-    const auto* eps = m_epsilon.data();
-    const auto* tmat = m_orientation.data();
+    const auto* forcing_vel = m_forcing_vel.data();
+    const auto* intrp_vel = m_intrp_vel.data();
+    const auto& sarr = m_ib_src(lev).array(mfi);
 
     amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-        const vs::Vector cc{
+        const vs::Vector ijk_x{
             problo[0] + (i + 0.5) * dx[0],
             problo[1] + (j + 0.5) * dx[1],
-            problo[2] + (k + 0.5) * dx[2],
+            problo[2] + (k + 0.5) * dx[2]
         };
 
-        amrex::Real src_force[AMREX_SPACEDIM]{0.0, 0.0, 0.0};
-        for (int ip = 0; ip < npts; ++ip) {
-            const auto dist = cc - pos[ip];
-            const auto dist_local = tmat[ip] & dist;
-            const auto dirac_delta_fac = utils::dirac_delta(dist_local, dx);
-            const auto& pforce = force[ip];
-
-            // FIXME: Is this the force definition for immersed boundaries ?
-            src_force[0] += dirac_delta_fac * pforce.x();
-            src_force[1] += dirac_delta_fac * pforce.y();
-            src_force[2] += dirac_delta_fac * pforce.z();
-        }
+        // loop over Lagrangian markers
+        for (int ip = 0; ip < m_pos.size(); ++ip) {
+            const auto lag_force = (forcing_vel[ip] - intrp_vel[ip]) / dt;
+
+            const auto dist_x = ijk_x - pos[ip];
+            const auto dirac_delta_fac = utils::dirac_delta(dist_x, ijk_x);
 
-        sarr(i, j, k, 0) += src_force[0];
```

Comment:
We need to keep src_force vector inside the loop  to do `src_force[0] += lag_force[0] * dirac_delta_fac * dV` and then add all contributions as `sarr(i, j, k, 0)=+= src_force[0];` outside the Lagrangian markers loop